migrate.py

```diff
@@ -507,10 +514,21 @@ def convert_changesets(content):
     Since they point to mercurial changesets and there's no easy way to map them
     to git hashes, better to remove them altogether.
     """
-    lines = content.splitlines()
-    filtered_lines = [l for l in lines if not l.startswith("→ <<cset")]
-    return "\n".join(filtered_lines)
-
+    if options.link_changesets:
+        # Look for things that look like sha's. If they are short, they must
+        # have a digit
+        def replace_changeset(match):
+            sha = match.group(1)
+            if len(sha) >= 8 or re.search(r"[0-9]", sha):
+                return ' [{sha} (bb)](https://bitbucket.org/{repo}/commits/{sha})'.format(
+                    repo=options.bitbucket_repo, sha=sha,
+                )
+        content = re.sub(r" ([a-f0-9]{6,40})\b", replace_changeset, content)
```

Comment:
been a while since I worked with regex's... what does the `{6,40}` here do? Is it replacing characters 6 through 40, or inspecting characters 6-40 or only matching strings of length 6-40?

Reply:
`[a-f0-9]{6,40}` means 6 to 40 hex characters.

Reply:
I tried to improve the heuristic by insisting that if the sha was shorter than 8 characters, it had to have a digit in it, so that English words wouldn't get linked.